### ISSUE TITLE
I expected the receipt field 'type' to be an int

### ISSUE THREAD

**Issue by carver** [ethereum/web3.py#2482]
* Version: 5.29.0
* Python: 3.8
* OS: osx/linux/win
* `pip freeze` output

### What was wrong?

```
>>> print(w3.eth.get_transaction_receipt(txn_hash).type)
"0x2"
```
I expected an `int`, in this case `2`

### How can it be fixed?

Add a field normalizer for receipt.type

**Comment by carver** [comment]
Also, I expected the `data` field in the logs to be `bytes`

**Comment by pacrob** [comment]
Can do, I'll get this tomorrow. Just to clarify, you want `bytes` and not `hexbytes`?

**Comment by carver** [comment]
> Can do, I'll get this tomorrow. Just to clarify, you want `bytes` and not `hexbytes`?

No `HexBytes` is even better!

**Comment by fselmo** [comment]
closed by #2488
